.github/workflows/shared-publish-to-docker-versioned.yaml

```diff
@@ -0,0 +1,116 @@
+name: Shared Pipeline to build and publish images to Docker
+on:
+  workflow_call:
+    inputs:
+      release_type:
+        description: The type of version number to return. Must be one of [Snapshot, Patch, Minor or Major]
+        required: true
+        type: string
+      publish_vulnerabilities:
+        description: If true, will attempt to publish any vulnerabilities to GitHub. Defaults to true. Set to false for private repos.
+        type: string
+        default: true
+      version_number_input:
+        description: The version number to publish for the docker repo
+        type: string
+        default: ''
+      force_release:
+        description: If 'yes', will force the creation a release, if 'no' will not create a release. 'branch' will use release_type and the branch to determine if a release should be created.
+        type: string
+        default: 'branch'
+      docker_file:
+        description: The Dockerfile used to build and publish Docker image
+        type: string
+        default: ''
+      docker_image_name:
+        description: The docker image name
+        required: true
+        type: string
+      docker_registry:
+        description: The docker registry
+        required: true
+        type: string
+env:
+  REGISTRY: ghcr.io
```

Comment:
You don't need both the ENV and the input. If they differ between repos, then leave it as an input, but otherwise use the ENV one.
Same applies to IMAGE_NAME